sqlmesh/core/engine_adapter/shared.py

```diff
@@ -101,7 +101,7 @@ def wrapper(*args: t.Any, **kwargs: t.Any) -> t.Any:
                 exp.to_table if obj_type == "TableName" else to_schema,
             )
             expression = to_expression_func(obj.copy() if isinstance(obj, exp.Table) else obj)
-            catalog_name = expression.args.get("catalog")
+            catalog_name = expression.catalog
```

Comment:
Bug I found while testing. This would be an identifier before but now will be a string (if it is defined).